browser/resources/settings/getting_started_page/brave_account_common.ts

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+export function onEyeIconClicked(event: Event) {
+  event.preventDefault()
+  let type = 'password'
+  const target = event.target as Element
+  target.setAttribute('name',
+    target.getAttribute('name') === 'eye-off'
+      ? (type = 'text', 'eye-on')
+      : 'eye-off')
+  target.parentElement!.setAttribute('type', type)
+}
+
+export function isEmailValid(input: string) {
+  if (!input) {
```

Comment:
If it's not already done elsewhere, we should strip leading/trailing whitespace before doing these checks.

Reply:
Right — the email regexp took care of whitespaces, but not the current solution. Instead of trimming, though, how about rejecting email addresses that contain the space character (the only whitespace in printable ASCII). That is, let's test for `/^[\x21-\x7F]+$/` instead of `/^[\x20-\x7F]+$/`.

Reply:
I think in this case what would be better from a user experience point of view would be to trim the email address entered by the user (leading/trailing whitespace is not uncommon when copy/pasting) so that it can pass our validation.

Reply:
Okay, let's trim the emai in `onEmailInput()` then. I'd still reject emails with spaces in them, though — WDYT? 🙂

Reply:
I don't think we need to reject email with spaces. The backend will do its own validation.